File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    answer = 0
    n = len(targets)

    targets.sort(key=lambda x: x[0])

    idx = n - 1

    while idx >= 0:
        start = targets[idx][0]

        while idx >= 0 and start < targets[idx][1]:
            idx -= 1

        answer += 1

    return answer

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.Arrays;

class Solution {
    
    public int solution(int[][] targets) {
        int answer = 0;
        int n = targets.length;
        
        Arrays.sort(targets, (a, b) -> Integer.compare(a[0], b[0]));
        
        int idx = n - 1;
        
        while (idx >= 0) {
            int start = targets[idx][0];
            while (idx >= 0 && start < targets[idx][1]) {
                idx--;
            }
            answer++;
        }
            
        return answer;
    }
}